datasets/sst/sst.py

```diff
@@ -0,0 +1,198 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{socher-etal-2013-recursive,
+    title = "Recursive Deep Models for Semantic Compositionality Over a Sentiment Treebank",
+    author = "Socher, Richard and Perelygin, Alex and Wu, Jean and
+      Chuang, Jason and Manning, Christopher D. and Ng, Andrew and Potts, Christopher",
+    booktitle = "Proceedings of the 2013 Conference on Empirical Methods in Natural Language Processing",
+    month = oct,
+    year = "2013",
+    address = "Seattle, Washington, USA",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/D13-1170",
+    pages = "1631--1642",
+}
+"""
+
+_DESCRIPTION = """\
+The Stanford Sentiment Treebank, the first corpus with fully labeled parse trees that allows for a
+complete analysis of the compositional effects of sentiment in language.
+"""
+
+_HOMEPAGE = "https://nlp.stanford.edu/sentiment/"
+
+_LICENSE = ""
+
+_URL = "https://www.dropbox.com/s/9j9dc55hs28wrye/stanfordSentimentTreebank.zip?dl=1"
```

Comment:
This is not the official download link right ?
Why not use the official link ?

Reply:
There is an [issue](https://groups.google.com/g/word2vec-toolkit/c/QIUjLw6RqFk/m/_iEeyt428wkJ) with the official files: `datasetSentences.txt` is almost impossible to decode. Every "special" symbol like â or ç results in an unreadable sentence whenever I look for its label in `dictionary.txt` and `sentiment_labels.txt`. The only solution I found was to manually replace all the special characters and save the resulting file in `utf-8`. 
Of course this is a bit patchy and I'd be happy for the data to be hosted on a more "official" server or, even better, find what encoding was used in the original file (which I kept in the repo as `datasetSentences_old_encoding.txt`. (You're welcome to give it a shot) 😜

Reply:
I think I tried almost all of the [Standard Encodings](https://docs.python.org/3/library/codecs.html#standard-encodings)

Reply:
I managed to read the old encoding this way:
```python
datasetSentences = open("datasetSentences.txt", encoding="utf-8")
datasetSentences_old_encoding = open("datasetSentences_old_encoding.txt", encoding="utf-8")

for a, b in zip(datasetSentences, datasetSentences_old_encoding):
    b = b.encode("utf-8").replace(b"\xc3\x83\xc2", b"\xc3").replace(b"\xc3\x82\xc2", b"\xc2").decode("utf-8")
    b = b.replace("-LRB-", "(").replace("-RRB-", ")")
    assert a == b
```
Using these two `replace` calls, you can read the original `datasetSentences.txt` file.

Reply:
🤦‍♂️ Of course! It had to be `\xc3\x83\xc2`! 🤣
Whenever you have time, let me know how did you get to that solution...

I'll go on and edit my code accordingly, thanks a lot for the fix 👍

Reply:
Haha yea that's really weird x)

I just compared the binary string of the the first badly encoded sentence with the corrected sentence and noticed that the only difference was these bad characters: `\x83\xc2` after `\xc3`. By chance removing these characters fixed all the encoding issues at once.

Reply:
Brilliant!